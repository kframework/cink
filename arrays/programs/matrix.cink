int main() {
  int a[2][3];
  a[1][2] = 7;
//  a[1] = {1, 2, 3};
  cout << a[1][2] << endl;
  return 0;
}
