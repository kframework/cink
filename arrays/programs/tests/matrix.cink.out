7
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        seq-array ( 2 , #symLoc(2) ) |-> array of 2 array of 3 int
        select ( seq-array ( 2 , #symLoc(2) ) , 1 ) |-> store ( const-array ( 
          3 , 0 ) , 2 , 7 )
        #symScalLoc(1) |-> lambda (  @ int ((a [ 2 ]) [ 3 ]) ; (((a [ 1 ]) [ 2
           ]) = 7) ; ((cout << ((a [ 1 ]) [ 2 ])) << (endl)) ; return 0 ; )
          
    </store>
</T>
